external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -134,7 +134,12 @@ public KafkaSpoutConfig(Builder<K, V> builder) {
         EARLIEST,
         LATEST,
         UNCOMMITTED_EARLIEST,
-        UNCOMMITTED_LATEST
+        UNCOMMITTED_LATEST;
+
+        @Override
+        public String toString() {
+            return "FirstPollOffsetStrategy{" + super.toString() + "}";
```

Comment:
This seems a little unnecessary, enums already have a good toString implementation?

Reply:
This toString() makes logging cleaner and more consistent. if you do log.debug("using {}",firstPollOffsetStrategy) it will print  "using FirstPollOffsetStrategy{EARLIEST}", which follows the pattern of every other toString() method. If we don't override toString() it will simply print "using EARLIEST".

Reply:
Makes sense.

Reply:
```
Seeking offset for topic-partition a1p-0 with FirstPollOffsetStrategy{UNCOMMITTED_EARLIEST} and OffsetAndMetadata{offset=3, metadata='{"topologyId":"tpn-1-1513534208","taskId":3,"threadName":"Thread-20-ks-executor[3 3]"}'}
```